homeassistant/components/mqtt/client.py

```diff
@@ -146,7 +146,9 @@ async def async_publish(
 
 
 def wrap_msg_callback(
```

Comment:
@epenet Any idea mypy says that L157 is unreachable. If I add `partial` as a type the issue is gone, but that feels incorrect.

Reply:
Reverted with # type: ignore[no-untyped-call]

Reply:
I think it's a bug from `mypy`, and you should add `# type: ignore[unreachable]`
cc @cdce8p

Reply:
You could type `msg_callback` with an additional `partial[None] | partial[Coroutine[Any, Any, None]`. The error should go away then. However that isn't really ideal either as it's not (yet) possible to type the arguments of a partial call.

I would also recommend to stick with `type: ignore[unreachable]` in this case.